modules/planning/tasks/deciders/path_bounds_decider/path_bounds_decider.cc

```diff
@@ -1316,29 +1316,28 @@ bool PathBoundsDecider::GetBoundaryFromStaticObstacles(
           } else {
             left_bounds.erase(left_bounds.find(curr_obstacle_l_min));
           }
-          obs_id_to_direction.erase(curr_obstacle_id);
-        }
-        // Update the bounds and center_line.
-        std::get<1>((*path_boundaries)[i]) = std::fmax(
-            std::get<1>((*path_boundaries)[i]),
-            *right_bounds.begin() + GetBufferBetweenADCCenterAndEdge());
-        std::get<2>((*path_boundaries)[i]) = std::fmin(
-            std::get<2>((*path_boundaries)[i]),
-            *left_bounds.begin() - GetBufferBetweenADCCenterAndEdge());
-        if (std::get<1>((*path_boundaries)[i]) >
-            std::get<2>((*path_boundaries)[i])) {
-          ADEBUG << "Path is blocked at s = " << curr_s;
-          path_blocked_idx = static_cast<int>(i);
-          if (!obs_id_to_direction.empty()) {
-            *blocking_obstacle_id = obs_id_to_direction.begin()->first;
+          obs_id_to_direction.erase(curr_obstacle_id);        
+          // Update the bounds and center_line.
+          std::get<1>((*path_boundaries)[i]) = std::fmax(
+              std::get<1>((*path_boundaries)[i]),
+              *right_bounds.begin() + GetBufferBetweenADCCenterAndEdge());
+          std::get<2>((*path_boundaries)[i]) = std::fmin(
+              std::get<2>((*path_boundaries)[i]),
+              *left_bounds.begin() - GetBufferBetweenADCCenterAndEdge());
+          if (std::get<1>((*path_boundaries)[i]) >
+              std::get<2>((*path_boundaries)[i])) {
+            ADEBUG << "Path is blocked at s = " << curr_s;
+            path_blocked_idx = static_cast<int>(i);
+            if (!obs_id_to_direction.empty()) {
+              *blocking_obstacle_id = obs_id_to_direction.begin()->first;
+            }
+            break;
+          } else {
+            center_line = (std::get<1>((*path_boundaries)[i]) +
+                          std::get<2>((*path_boundaries)[i])) /
+                          2.0;
           }
```

Comment:
can you explain little bit why u want to move this chunk of code inside else of line 1316?

Reply:
when a new obstacle enters into the scope, line 1293 and line 1304 updates the boundary using " UpdatePathBoundaryAndCenterLine()". However, line 1319- 1333 updates the boundary again if this chunk of code is not moved inside "else" of line 1316. I supoose updating the boundary twice is unnecessary.

Reply:
I'll leave this to @panjiacheng to evaluate. thx

Reply:
@henryhu6 can you please fix the CB failures first?

Reply:
what do CB failures mean?

Reply:
Continuous Build. scroll down this page you can see it. 
"Build (Apollo) - merge — TeamCity build failed"